pkg/openstack/loadbalancer.go

```diff
@@ -727,25 +727,52 @@ func getSubnetIDForLB(network *gophercloud.ServiceClient, node corev1.Node, pref
 	return "", cpoerrors.ErrNotFound
 }
 
+// isPortMember returns true if IP and subnetID are one of the FixedIPs on the port
+func isPortMember(port PortWithPortSecurity, IP string, subnetID string) bool {
+	for _, fixedIP := range port.FixedIPs {
+		if (subnetID == "" || subnetID == fixedIP.SubnetID) && IP == fixedIP.IPAddress {
+			return true
+		}
+	}
+	return false
+}
+
 // applyNodeSecurityGroupIDForLB associates the security group with all the ports on the nodes.
-func applyNodeSecurityGroupIDForLB(network *gophercloud.ServiceClient, nodes []*corev1.Node, sg string) error {
+func applyNodeSecurityGroupIDForLB(network *gophercloud.ServiceClient, svcConf *serviceConfig, nodes []*corev1.Node, sg string) error {
 	for _, node := range nodes {
 		serverID, _, err := instanceIDFromProviderID(node.Spec.ProviderID)
 		if err != nil {
 			return fmt.Errorf("error getting server ID from the node: %w", err)
 		}
+
+		addr, err := nodeAddressForLB(node, svcConf.preferredIPFamily)
+		if err != nil {
+			// If node has no viable address let's ignore it.
+			continue
+		}
+
 		listOpts := neutronports.ListOpts{DeviceID: serverID}
-		allPorts, err := openstackutil.GetPorts(network, listOpts)
+		allPorts, err := openstackutil.GetPorts[PortWithPortSecurity](network, listOpts)
```

Comment:
um... where's the usage of this ? I only see it's used but where's this created?

Reply:
What do you mean? If where `allPorts` var is used, then in for loop just below. And to have `PortSecurityEnabled` property on ports I need to use `PortWithPortSecurity` struct and that's why I made `GetPorts()` a generic.

Reply:
sorry, I didn't notice change in
https://github.com/kubernetes/cloud-provider-openstack/pull/2455/files#diff-a89aa055a2316c0db33d258ec1c17fc3948b8f53fbc3a475d5a6602025096b33L143
at first so didn't get the logic here, but read that code more I understand the purpose , thanks for the info